components/brave_wallet_ui/components/desktop/hardware-wallet-connect/hardware_wallet_connect.tsx

```diff
@@ -117,205 +166,163 @@ export const HardwareWalletConnect = ({
+      return
+    }
+    setAccounts([])
+    setTotalNumberOfAccounts(DerivationBatchSize)
+    setCurrentDerivationScheme(scheme)
+  }
 
-  const onAddHardwareAccounts = React.useCallback(
-    async (accounts: BraveWallet.HardwareWalletAccount[]) => {
-      try {
-        await importHardwareAccounts(accounts).unwrap()
-        onSuccess()
-      } catch (error) {
-        console.log(error)
+  const onAccountChecked = (path: string, checked: boolean) => {
+    const newAccounts = accounts.map((acc) => {
+      if (acc.derivationPath === path) {
+        return { ...acc, shouldAddToWallet: checked }
       }
-    },
-    [onSuccess, importHardwareAccounts]
-  )
+      return acc
+    })
+    setAccounts(newAccounts)
+  }
 
-  const onChangeDerivationScheme = React.useCallback(
-    (scheme: HardwareDerivationScheme) => {
-      if (!selectedHardwareWallet) return
-
-      setSelectedDerivationScheme(scheme)
-      setAccounts([])
-      onConnectHardwareWallet({
-        hardware: selectedHardwareWallet,
-        startIndex: 0,
-        stopIndex: DerivationBatchSize,
-        scheme: scheme,
-        coin: selectedAccountType.coin,
-        network: filecoinNetwork,
-        onAuthorized: hideAuthorizeDevice
-      })
-        .then((result) => {
-          setAccounts(result)
-        })
-        .catch((error) => {
-          setConnectionError(getErrorMessage(error, selectedAccountType.name))
-          setShowAccountsList(false)
+  const loadMoreAccounts = React.useCallback(async () => {
+    const numberOfAccountsToLoad = totalNumberOfAccounts - accounts.length
+    if (numberOfAccountsToLoad <= 0) {
+      return
+    }
+
+    setIsConnecting(true)
+    const result = await loadAccountsFromDevice({
+      startIndex: accounts.length,
+      count: numberOfAccountsToLoad,
+      scheme: currentHardwareImportScheme,
+      onAuthorized: hideAuthorizeDevice
+    })
+    setIsConnecting(false)
+
+    if (result.payload) {
+      setShowAccountsList(true)
+
+      const newAccounts: Array<Required<AccountFromDevice>> =
+        result.payload.map((acc) => {
+          const alreadyInWallet = !!savedAccounts.find((a) => {
+            return (
+              a.accountId.kind === BraveWallet.AccountKind.kHardware &&
+              a.address === acc.address
+            )
+          })
+          return { ...acc, alreadyInWallet, shouldAddToWallet: false }
         })
-        .finally(() => setIsConnecting(false))
-    },
-    [selectedHardwareWallet, selectedAccountType, filecoinNetwork]
-  )
 
-  const getDefaultAccountName = React.useCallback(
-    (account: BraveWallet.HardwareWalletAccount) => {
-      const index = accounts.findIndex((e) => e.address === account.address)
-      let schemeString
-      switch (selectedDerivationScheme) {
-        case LedgerDerivationPaths.Legacy:
-          schemeString = ' (Legacy)'
-          break
-        default:
-          schemeString = ''
+      setAccounts([...accounts, ...newAccounts])
```

Comment:
Can we do this instead so that we can replace `accounts` in the deps array with `accounts.length` ?
```suggestion
      setAccounts(prev => prev.concat(newAccounts))
```